playground/terraform/README.md

```diff
@@ -16,74 +16,117 @@
+region               = "us-east1"            #Set the deployment region
+location             = "us-east1-b"          #Select the deployment location from available in the specified region
+state_bucket         = "bucket_name"         #GCS bucket name for Beam Playground temp files
+bucket_examples_name = "bucket_name-example" #GCS bucket name for Playground examples storage
 ```
-
-* Backend config
-
-```bash
-echo 'bucket = "put your state bucket name here"' > /path/to/beam/playground/terraform/environment/{env-name}/state.tfbackend
+* `state.tfbackend` environment variables:
 ```
-
-* Terraform variables config and provide necessary variables
-
-```bash
-touch /path/to/beam/playground/terraform/environment/{env-name}/terraform.tfvars
+bucket               = "bucket_name"         #input bucket name - will be used for terraform tfstate file
 ```
-
-Then provide necessary variables.
-
-## 2. Provision infrastructure
-
-To deploy Playground infrastructure run gradle task:
-
-```bash
-./gradlew playground:terraform:InitInfrastructure -Pproject_environment="env-name"
+2. Configure authentication for the Google Cloud Platform
 ```
-
-## 3. Deploy application
-
-To deploy application run following steps:
-
-* Authinticate in Artifact registry
-
-```bash
-gcloud auth configure-docker us-central1-docker.pkg.dev
+gcloud init
+```
+```
+gcloud auth application-default login
 ```
+# Deploy Playground infrastructure:
+1. Start the following command from the top level repository folder ("beam") to deploy the Payground infrastructure:
+```
+./gradlew playground:terraform:InitInfrastructure -Pproject_environment="environment_name" -Pdns-name="playground.zone"
+```
+Where playground.zone - chosen DNS for Playground
 
-* Вeploy backend services
+2. Find a Static IP in your GCP project>VPC Network>IP Addresses>pg-static-ip
+<br>Add following DNS A records for the discovered static IP address:
+```
+java.playground.zone
+python.playground.zone
+scio.playground.zone
+go.playground.zone
+router.playground.zone
+playground.zone
+```
+Where "playground.zone" is the registered DNS zone<br>
+[More about DNS zone registration](https://domains.google/get-started/domain-search/)<br>
+[More about A records in DNS](https://support.google.com/a/answer/2579934?hl=en)
 
-```bash
-./gradlew playground:terraform:deployBackend -Pproject_environment="env-name" -Pdocker-tag="tag"
+# Deploy Playground to Kubernetes:
+1. Run the following command to authenticate in the Docker registry:
 ```
+ gcloud auth configure-docker `chosen_region`-docker.pkg.dev
+```
+2. Run the following command to authenticate in GKE:
+```
+gcloud container clusters get-credentials --region `chosen_location` `gke_name` --project `project_id`
+```
+Start the following command from the top level repository folder ("beam") to deploy the Payground infrastructure:
+```
+./gradlew playground:terraform:gkebackend -Pproject_environment="environment_name" -Pdocker-tag="tag" -Pdns-name="playground.zone"
```

Comment:
When I ran this command, I received the error:
```
> Task :playground:frontend:docker FAILED

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':playground:terraform:helmInstallPlaygroundToDefault'.
> There was a failure while executing work items
   > A failure occurred while executing org.unbrokendome.gradle.plugins.helm.command.HelmExecWorkAction
      > A problem occurred starting process 'command 'helm''
```

I run helm version, I see:

```
version.BuildInfo{Version:"v3.9.4", GitCommit:"dbc6d8e20fe1d58d50e6ed30f09a04a77e4c68db", GitTreeState:"clean", GoVersion:"go1.17.13"}
```

Reply:
So far, I think I've isolated that at least when I executed the `./gradlew playground:terraform:gkebackend` command, that the problem wasn't with helm but with the building of the docker images.  I was able to execute the `./gradlew :playground:backend:containers:scio:dockerPush` successfully and see in my Google Cloud project successful deployment so far of each successfully built image.